### ISSUE TITLE
This code crashes every time

### ISSUE THREAD

**Issue by kyriosli** [laverdet/node-fibers#282]
I ran into a problem that my server crashes randomly and then simplified my code as follows:

```js
"use strict"

const Fibers = require('./node-fibers');

function walk_through(str) {
    for (let i = 0, len = str.length; i < len; i++) {
        str[i];
    }
}

const fiber = new Fibers(function() {
    for(let i = 0; ; i++) {
        console.log(i);

        walk_through('0');
    }
});

fiber.run();
```

Then I ran it with `node --harmony test.js`, and it crashes every time:

```
...
4915
4916
4917


#
# Fatal error in ../deps/v8/src/lookup.cc, line 47
# CHECK(!root->IsNull()) failed
#

==== C stack trace ===============================

 1: V8_Fatal
 2: v8::internal::LookupIterator::GetRoot() const
 3: v8::internal::LoadIC::Load(v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Name>)
 4: v8::internal::LoadIC_Miss(int, v8::internal::Object**, v8::internal::Isolate*)
 5: ??
Illegal instruction
```
Tested:

  - windows: fail with node-0.12.7, ok with node-5.1.0
  - linux: fail with node-0.12.7

I found out some solution such as using `var` instead of `let` in line `for(let i = 0, len = ...`.  Maybe it's a bug of nodejs or v8 and only shows up in older version of it?

**Comment by laverdet** [comment]
Would you mind testing this on a modern version of node (v4 or v5) and let me know if it fails for you? There's been a lot of threading fixes in v8 since node 0.12.x.

**Comment by kyriosli** [comment]
Oops, I thought it had nothing to do with `fibers`.

I tried again yesterday to run the code without `fibers`, and it crashes too:

```js
"use strict"

function walk_through(str) {
    for (let i = 0, len = str.length; i < len; i++) {
        str[i];
    }
}

(function() {
    for(let i = 0; ; i++) {
        console.log(i);

        walk_through('0');
    }
})()
```

So I think maybe it's a problem of `--harmony` that has already been solved in latter versions of `node`.
